homeassistant/components/netgear/router.py

```diff
@@ -0,0 +1,209 @@
+"""Represent the Netgear router and its devices."""
+from datetime import timedelta
+import logging
+
+from pynetgear import Netgear
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_URL,
+    CONF_USERNAME,
+)
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.device_registry import format_mac
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.util import dt as dt_util
+
+from .const import (
+    CONF_CONSIDER_HOME,
+    CONF_TRACKED_LIST,
+    DEFAULT_CONSIDER_HOME,
+    DOMAIN,
+    MODELS_V2,
+)
+from .errors import CannotLoginException
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def convert_tracked_list(tracked_list_str):
+    """Convert tracked list string to a list."""
+    tracked_list = []
+    tracked_list_unformatted = []
+    if tracked_list_str:
+        tracked_list_unformatted = (
+            tracked_list_str.replace(" ", "")
+            .replace("]", "")
+            .replace("[", "")
+            .split(",")
+        )
```

Comment:
This looks a bit brittle

Reply:
yea I am not perfectly happy with it eather, but as far as I know there is no good way to implement a user specifiable list as an options parameter.
So now I just have a comma seperated string where users can put in there list of mac adresses.
But to be a bit more flexable for users I first remove all spaces and [ and ] chars in case a users enters those, then I split on the comma's.

Is there a propper way to let users specify a list of strings (mac's) with variable lenght as an option variable?

Reply:
Maybe pass it though `cv.ensure_list_csv` and then clean up each item?

Reply:
good idea, thanks, did that